mypy/semanal.py

```diff
@@ -1600,6 +1594,25 @@ def configure_base_classes(self,
             return
         self.calculate_class_mro(defn, self.object_type)
 
+    def enum_has_final_values(self, base: Instance) -> bool:
+        if (
+            base.type.is_enum
+            and base.type.fullname not in (
+                'enum.Enum', 'enum.IntEnum', 'enum.Flag', 'enum.IntFlag')
+            and base.type.names
+            and base.type.defn
+        ):
+            # TODO: if `Var` ever has `.has_explicit_value` prop, remove `ast` traversal.
+            # Check that enum members are really initialized, ignore methods.
+            # We need at least one assigned member to call this Enum `final`.
+            for node in base.type.defn.defs.body:
+                if isinstance(node, (FuncBase, Decorator)):
+                    continue  # A method
+                if isinstance(node, AssignmentStmt) and isinstance(node.rvalue, TempNode):
```

Comment:
should this also check `node.rvalue.no_rhs` like https://github.com/python/mypy/blob/7f0ad943e4b189224f2fedf43aa7b38f53ec561e/mypy/semanal.py#L2412

Reply:
> but this still doesn't fix the second two errors from #11578 (comment), right?

Yes, see this comment: https://github.com/python/mypy/pull/11579#discussion_r753334973


> should this also check node.rvalue.no_rhs like 

Done! Thanks.